cl_hubeau/water_services/utils.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Convenience functions for water services indicators.
+"""
+
+import geopandas as gpd
+import pandas as pd
+from tqdm import tqdm
+
+from cl_hubeau.water_services.water_services_scraper import WaterServicesSession
+from cl_hubeau import _config
+from cl_hubeau.utils import get_cities
+
+
+def get_all_communes(**kwargs) -> gpd.GeoDataFrame:
+    """
+    Gets the services for every french city.
+
+    Parameters
+    ----------
+    **kwargs :
+        kwargs passed to WaterServicesSession.get_communes (hence mostly intended
+        for hub'eau API's arguments). Do not use `format` or `code_commune`
+        as they are set by the current function.
+
+    Returns
+    -------
+    results : gpd.GeoDataFrame
+        GeoDataFrame of water services indicators for every french city.
+
+    """
+    with WaterServicesSession() as session :
+        cities = get_cities()
+        results = [
+            session.get_communes(
+                code_commune=city, format="geojson", **kwargs
+            )
+            for city in tqdm(
+                cities,
+                desc="querying city/city",
+                leave=_config["TQDM_LEAVE"],
+                position=tqdm._get_free_pos(),
+            )
+        ]
```

Comment:
This is going to be particularly slow (36k cities in France). Did you test it with a loop over departements?